Lacrit/assignment4/islands.cpp

```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+ 
+const int rows = 5;
+const int cols = 5;
+
+// A function to check if a given cell (row, col) can be included in DFS
+int isValid(int mat[rows][cols], int row, int col, bool visited[rows][cols])
```

Comment:
It will be better if the code can have matrices of different sizes. This could be done having three parameters: `int **mat, int num_rows, int num_cols`, or preferably using a `vector<vector<int>>`.
For example, this function could become:
```c++
bool isValid(const vector<vector<bool>> &mat, int row, int col, const vector<vector<bool>> &visited) 
{
    // row number is in range, column number is in range and value is 1 
    // and not yet visited
    return (row >= 0) && (row < mat.size()) &&     
           (col >= 0) && (col < mat[row].size()) &&      
           (mat[row][col] && !visited[row][col]); 
}
```
Marking them as a `const &` is not strictly needed, but without them the compiler will make a copy of the matrices every time we call this functions, which is a bit slow.